components/supervisor/pkg/ports/ports.go

```diff
@@ -191,10 +191,21 @@ func (pm *Manager) updateState(exposed []ExposedPort, served []ServedPort, confi
 	if exposed != nil && !reflect.DeepEqual(pm.exposed, exposed) {
 		pm.exposed = exposed
 	}
-	if served != nil && !reflect.DeepEqual(pm.served, served) {
-		pm.served = served
-		pm.updateProxies()
+
+	if served != nil {
+		newServed := make(map[uint32]ServedPort)
+		for _, port := range served {
+			current, exists := newServed[port.Port]
+			if !exists || (!port.BoundToLocalhost && current.BoundToLocalhost) {
+				newServed[port.Port] = port
+			}
+		}
+		if !reflect.DeepEqual(pm.served, newServed) {
```

Comment:
Do we need this check? If we don't do it, we might just run `updateProxies` again without it actually doing anything, no?

I'd reckon that's faster anyways than `reflect.DeepEqual`.

Reply:
It would also produce more status updates then, i.e. each 2s without real changes.